### PR TITLE
Fix broken test in python3

### PR DESCRIPTION
**Pull request by zachmullen**
A recent change in httmock behavior broke testing in python 3.
This also fixes several warnings from unclosed file handles.

### CODE REVIEW

File: tests/cases/py_client/lib_test.py [hunk 1]
```diff
@@ -517,7 +518,7 @@ def testPatch(self):
         }
 
         def _patchJson(url, request):
-            patchRequest['valid'] = json.loads(request.body) == jsonBody
+            patchRequest['valid'] = json.loads(request.body.decode('utf8')) == jsonBody
```
Comment (zachmullen) [inline tests/cases/py_client/lib_test.py]: This is the line that contains the fix of the failing test.

**Comment by zachmullen** [comment]
@girder/developers could someone PTAL at this, it's blocking our CI.

**Comment by cpatrick** [comment]
LGTM.

**Comment by cpatrick** [comment]
🚀 🚢 :shipit:

**Pull request merged**